### ISSUE TITLE
Run Wasmer and Wasmtime together

### ISSUE THREAD

**Issue by maxzaver** [near/nearcore#3187]
To ensure there are no Wasmer- or Wasmtime-specific bugs in our codebase we need to run both Wasmer and Wasmtime side-by-side and panic when they disagree.

Specifically, each block is going to be produced using Wasmer and Wasmtime executed in parallel. Each of them would produce a different `TrieUpdate` and other artifacts, like execution outcome and receipts. Before we call `TrieUpdate::finalize` we would compare them and if they are different we would panic. This will guarantee that if there is a transient error caused by Wasmer/Wasmtime the block producer is not going to be slashed. This would also guarantee that if there is a deterministic error that happens only on some nodes the network does not diverge.

This however, would require increasing hardware requirements 2x for our validators. CC @jimmy3dita , @bowenwang1996 , @chefsale

**Comment by jimmy3dita** [comment]
Is this a long-term requirement, or is needed by now to provide additional stability?

**Comment by olonho** [comment]
It is medium term goal.

**Comment by nearmax** [comment]
We should try to enable it in the next 1-2 months, to ensure extra layer of safety.

**Comment by olonho** [comment]
OK, will take a look on feasibility of this one.

**Comment by bowenwang1996** [comment]
@olonho are we still working on this issue?

**Comment by bowenwang1996** [comment]
@matklad do we still plan to do this?

**Comment by matklad** [comment]
Yeah, we do want to do that eventually, but its no-longer on the near-term roadmap.

**Comment by matklad** [comment]
Three thoughts here: 

* It doesn't seem likely that we'll be able to run two vms in prod any time soon, for various technical reasons
* We got most of the benefit here via differential fuzzing of wasmtime vs wasmer I think. @Ekleog-NEAR, what's our feeling on that? Do we get enough coverage here?
* What we can potentially do is setup wasmtime canary nodes, or history replay.

**Comment by Ekleog** [comment]
I also think that we get more benefit from differential fuzzing of wasmer vs wasmtime than we would from panicking if wasmer and wasmtime were to disagree: I wouldn't be surprised if there were an instance where wasmer and wasmtime disagree (even though fuzzing didn't find anything yet), and panicking all the nodes in the network at once would be very bad if an attacker discovered one such case.

That said, having wasmtime canary node might be a good idea, if only to get real-world examples of discrepancies, if there is any discrepancy to be found, so we know about them and can fix them.
